test/vmtests/vmtests/test_imagecreator.py

```diff
@@ -0,0 +1,434 @@
+    logging.info(f"LibVirt VM object created")
+
+    # Start VM.
+    logging.info(f"Starting VM...")
+    vm.start()
+    logging.info(f"VM started successfully!")
+
+    # Connect to the VM.
+    logging.info(f"Attempting to connect to VM via SSH...")
+    logging.info(f"Username: {username}")
+
+    with vm.create_ssh_client(ssh_private_key_path, test_temp_dir, username) as ssh_client:
+        logging.info(f"SSH connection established successfully!")
+        # Run the test
+        logging.info(f"Running basic checks on the VM...")
+        run_basic_checks(ssh_client, test_temp_dir)
+        logging.info(f"Basic checks completed successfully!")
+
+
+def run_basic_checks(
+    ssh_client: SshClient,
+    test_temp_dir: Path,
+) -> None:
+
+    ssh_client.run("cat /proc/cmdline").check_exit_code()
+
+    os_release_path = test_temp_dir.joinpath("os-release")
+    ssh_client.get_file(Path("/etc/os-release"), os_release_path)
+
+    with open(os_release_path, "r") as os_release_fd:
+        os_release_text = os_release_fd.read()
+
+        # Since imagecreator creates new images, we expect Azure Linux 3.0
+        assert "ID=azurelinux" in os_release_text
+        assert 'VERSION_ID="3.0"' in os_release_text
+
+    # Check that essential packages are installed
+    ssh_client.run("rpm -q kernel").check_exit_code()
+    ssh_client.run("rpm -q systemd").check_exit_code()
+    ssh_client.run("rpm -q grub2").check_exit_code()
+    ssh_client.run("rpm -q bash").check_exit_code()
+
+
+@pytest.mark.skipif(platform.machine() != "x86_64", reason="arm64 is not supported for this combination")
+def test_create_image_efi_qcow_output(
+    image_creator_binary_path: Path,
+    rpm_sources: List[Path],
+    tools_tar: Path,
+    ssh_key: Tuple[str, Path],
+    test_temp_dir: Path,
+    test_instance_name: str,
+    logs_dir: Path,
+    libvirt_conn: libvirt.virConnect,
+    close_list: List[Closeable],
+    image_customizer_binary_path: Path,
+) -> None:
+    config_path = IMAGECREATOR_TEST_CONFIGS_DIR.joinpath("minimal-os.yaml")
+    output_format = "qcow2"
+    # debug message
+    logging.debug("Running test_create_image_efi_qcow_output...")
+
+    run_image_creator_test(
+        image_creator_binary_path,
+        rpm_sources,
+        tools_tar,
+        config_path,
+        output_format,
+        ssh_key,
+        test_temp_dir,
+        test_instance_name,
+        logs_dir,
+        libvirt_conn,
+        close_list,
+        image_customizer_binary_path,
+    )
+
+
+def test_create_image_efi_raw_output(
+    image_creator_binary_path: Path,
+    rpm_sources: List[Path],
+    tools_tar: Path,
+    ssh_key: Tuple[str, Path],
+    test_temp_dir: Path,
+    test_instance_name: str,
+    logs_dir: Path,
+    libvirt_conn: libvirt.virConnect,
+    close_list: List[Closeable],
+    image_customizer_binary_path: Path,
+) -> None:
+    config_path = IMAGECREATOR_TEST_CONFIGS_DIR.joinpath("minimal-os.yaml")
+    output_format = "raw"
```

Comment:
Outputting `raw` vs. `qcow2` is a fairly difference in the behavior of imagecreator. So, I don't think it is worth having an additional VMTests test for it.